src/Control/Carrier/FossaApiClient/Internal/FossaAPIV1.hs

```diff
@@ -1148,10 +1151,23 @@ vsiAddFilesToScan :: (Has (Lift IO) sig m, Has Diagnostics sig m) => ApiOpts ->
 vsiAddFilesToScan apiOpts scanID files = fossaReq $ do
   (baseUrl, baseOpts) <- useApiOpts apiOpts
 
-  let body = VSIAddFilesToScanRequestBody files
+  -- This is in the API layer because it's an implementation detail of the API.
+  -- The API expects all file paths to be normalized to forward slashes.
+  let normalized = Map.fromList . normalizeFiles $ Map.toList files
+  let body = VSIAddFilesToScanRequestBody normalized
   _ <- req POST (vsiAddFilesToScanEndpoint baseUrl scanID) (ReqBodyJsonCompat body) ignoreResponse baseOpts
 
   pure ()
+  where
+    normalizeFiles :: [(Path Rel File, Fingerprint.Combined)] -> [(FilePath, Fingerprint.Combined)]
+    normalizeFiles = fmap normalizeFile
+
+    normalizeFile :: (Path Rel File, Fingerprint.Combined) -> (FilePath, Fingerprint.Combined)
+    normalizeFile (path, fp) = (normalizePath path, fp)
+
+    normalizePath :: Path Rel File -> FilePath
+    normalizePath input | pathSeparator == '/' = Path.toFilePath input
+    normalizePath input = toString . Text.intercalate "/" $ toText <$> splitDirectories (Path.toFilePath input)
```

Comment:
According to this, you can still have `\` in the list entries after running `splitDirectories`: 
https://hackage.haskell.org/package/filepath-1.4.100.0/docs/System-FilePath-Posix.html#v:splitDirectories

```
Windows: splitDirectories "C:\\test\\file" == ["C:\\", "test", "file"]
Windows: splitDirectories "C:\\test\\\\\\file" == ["C:\\", "test", "file"]
```

Reply:
I'm not sure if this is a problem that we can encounter, though. It looks like it doesn't happen for relative paths and we might only be encountering relative paths here?

Reply:
> unless we're certain that we're always using relative paths and this isn't a problem for relative paths

We are; the parent function is this:
```haskell
vsiAddFilesToScan :: (Has (Lift IO) sig m, Has Diagnostics sig m) => ApiOpts -> VSI.ScanID -> Map (Path Rel File) Fingerprint.Combined -> m ()
```

So we are enforcing that we get relative files via types.

Reply:
Perfect. Thanks!